vic/drivers/shared_all/src/calc_root_fraction.c

```diff
@@ -41,128 +41,126 @@ calc_root_fractions(veg_con_struct  *veg_con,
     int                  veg;
     size_t               layer;
     size_t               zone;
-    size_t               i;
-    size_t               n_iter;
+    size_t               ltmp;
+    double               Lsum;
+    double               Lsumprev;
+    double               Zsum;
+    double               Dsum;
+    double               Dsumprev;
+    double               dD;
     double               sum_fract;
+    double              *root_dens;
     double               dum;
-    double               Zstep;
-    double               Zsum;
-    double               Lstep;
-    double               Lsum;
-    double               Zmin_fract;
-    double               Zmin_depth;
-    double               Zmax;
+
+    root_dens = calloc(options.ROOT_ZONES, sizeof(double));
 
     Nveg = veg_con[0].vegetat_type_num;
 
     for (veg = 0; veg < Nveg; veg++) {
-        sum_fract = 0;
+
+        // Compute density of root fractions in each root zone
+        for (zone = 0; zone < options.ROOT_ZONES; zone++) {
+            if (veg_con[veg].zone_depth[zone] > 0) {
+                root_dens[zone] = veg_con[veg].zone_fract[zone] /
+                                  veg_con[veg].zone_depth[zone];
+            }
+            else {
+                root_dens[zone] = 1.0;
```

Comment:
Is there a reason why setting `root_dens[zone] = 1.0` in the case that `zone_depth[zone] == 0`? (I guess in practice, no matter what value `root_dens[zone]` is, it will eventually be multiplied by zero so it doesn't really matter)

Reply:
Exactly, the value doesn't matter.  I like `1.0` because if you ever want to take the inverse, it is defined.